Update umb-tree-item.less

mistyn8 opened the pull request:
Icon sizing inconsistencies in tree

this fixes https://github.com/umbraco/Umbraco-CMS/issues/9699

nathanwoulfe commented:
@mistyn8 could you please double check this - adding `flex-shrink:0` to `.umb-tree-item` won't solve this problem as it's not a flex element. I think it needs to be on `.umb-tree-icon` as that's the element we want to prevent shrinking...

mistyn8 commented:
@nathanwoulfe you aren't wrong. Sorry that the initial commit didn't reflect the solution posted on https://github.com/umbraco/Umbraco-CMS/issues/9699

Updated the PR so `flex-shrink:0` now added to `.umb-tree-icon` instead.

nathanwoulfe reviewed (approved):
All good, thanks @mistyn8
